GillianCore/logging/report.ml

```diff
@@ -0,0 +1,50 @@
+type severity = Info | Log | Success | Error | Warning
+
+type 'a t = {
+  id : Uuidm.t;
+  title : string;
+  elapsed_time : float;
+  previous : Uuidm.t option;
+  parent : Uuidm.t option;
+  content : (('a, Format.formatter, unit) format -> 'a) -> unit;
+  severity : severity;
+}
+
+let parents : Uuidm.t Stack.t = Stack.create ()
+
+let current : Uuidm.t option ref = ref Option.none
+
+let seed = Random.State.make_self_init ()
+
+type 'a report_builder = unit -> 'a t
+
+let report_builder title content severity (_unit : unit) =
```

Comment:
```suggestion
let make ~severity ~title ~content () =
```
This is closer to the OCaml conventions. See things like [ppx_make](https://github.com/ocaml-ppx/ppx_deriving#plugin-make)